src/cunumeric/index/putmask_template.inl

```diff
@@ -0,0 +1,125 @@
+/* Copyright 2022 NVIDIA Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#pragma once
+
+// Useful for IDEs
+#include <core/utilities/typedefs.h>
+#include "cunumeric/index/putmask.h"
+#include "cunumeric/pitches.h"
+#include "cunumeric/execution_policy/indexing/bool_mask.h"
+
+namespace cunumeric {
+
+using namespace Legion;
+using namespace legate;
+
+template <VariantKind KIND, LegateTypeCode CODE, int DIM, int VDIM, bool SCALAR_VALUE = false>
```

Comment:
like I said in the other comment, if you do numpy broadcasting on the values, you can remove the `SCALAR_VALUE == true` case.